src/main/java/org/aigents/nlp/gen/Segment.java

```diff
@@ -99,17 +118,51 @@ private static boolean check(String[] input) {
 			return false;
 		if (last.equals("on") || first.equals("on"))
 			return false;
-		if (last.equals("with") || first.equals("with"))
-			return false;
-		if (first.equals("board"))
+		if (last.equals("with") || last.equals("the") || first.equals("with"))
```

Comment:
What do we need to hardcode the language-specific hacks right in the generic code. The algorithm should rely on the Links Grammar, not on the grammatical rules hardcoded in the code